pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -204,3 +170,117 @@ func podWithUpdatedSelectionElements(pod *corev1.Pod, networks []v1.NetworkSelec
 	newPod.Annotations[v1.NetworkAttachmentAnnot] = string(newNets)
 	return newPod, nil
 }
+
+func ensureIPAMClaimRefAtNetworkSelectionElements(ctx context.Context,
+	cli client.Client, pod *corev1.Pod, vmName string,
+	networkSelectionElements []*v1.NetworkSelectionElement) (changed bool, response string, err error) {
+	log := logf.FromContext(ctx)
+	hasChangedNetworkSelectionElements := false
+	for i, networkSelectionElement := range networkSelectionElements {
+		nadName := types.NamespacedName{
+			Namespace: networkSelectionElement.Namespace,
+			Name:      networkSelectionElement.Name,
+		}.String()
```

Comment:
nit: wouldn't it be simpler to %s.%s ? Don't we have an helper for that ? 

at least compose the `nadKey` before, and here you can just do `nadKey.String()`